spec/rubocop/cop/rails/active_model_errors_direct_manipulation_spec.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+RSpec.describe RuboCop::Cop::Rails::ActiveModelErrorsDirectManipulation, :config do
+  let(:message) { 'Avoid manipulating ActiveModel errors as hash directly.' }
+
+  context 'when modifying errors' do
+    it 'registers an offense' do
+      expect_offense <<~RUBY
+        user.errors[:name] << 'msg'
+        ^^^^^^^^^^^^^^^^^^^^^^^^^^^ #{message}
+      RUBY
+    end
+
+    context 'when assigning' do
+      it 'registers an offense' do
+        expect_offense <<~RUBY
+          user.errors[:name] = []
+          ^^^^^^^^^^^^^^^^^^^^^^^ #{message}
+        RUBY
+      end
+    end
+  end
+
+  context 'when modifying errors.messages' do
+    it 'registers an offense' do
+      expect_offense <<~RUBY
+        user.errors.messages[:name] << 'msg'
+        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ #{message}
+      RUBY
+    end
+
+    context 'when assigning' do
+      it 'registers an offense' do
+        expect_offense <<~RUBY
+          user.errors.messages[:name] = []
+          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ #{message}
+        RUBY
+      end
+    end
+  end
+
+  context 'when modifying errors.details' do
+    it 'registers an offense' do
+      expect_offense <<~RUBY
+        user.errors.details[:name] << {}
+        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ #{message}
+      RUBY
+    end
+
+    context 'when assigning' do
+      it 'registers an offense' do
+        expect_offense <<~RUBY
+          user.errors.details[:name] = []
+          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ #{message}
+        RUBY
+      end
+    end
+  end
```

Comment:
It may be acceptable if receiver is not specified:
`errors[:name] << 'msg'`

Can you add the test case?

Reply:
Now I think about it, I feel it is important if the receiver is not specified. However it could leave to many false positives. I want to find a way so this receiver-less check can be done only to files under the `models` directory. I wonder if that is possible.